lib/components/modal.vue

```diff
@@ -249,49 +249,61 @@
             modalClasses() {
                 return [
                     'modal',
-                    this.noFade ? '' : 'fade',
-                    this.is_show ? 'show' : '',
-                    this.is_block ? 'd-block' : ''
+                    {
+                        fade: !this.noFade,
+                        show: this.is_show,
+                        'd-block': this.is_block
+                    }
                 ];
             },
             dialogClasses() {
                 return [
                     'modal-dialog',
-                    Boolean(this.size) ? `modal-${this.size}` : ''
+                    {
+                        [`modal-${this.size}`]: this.size
+                    }
                 ];
             },
             backdropClasses() {
                 return [
                     'modal-backdrop',
-                    this.noFade ? '' : 'fade',
-                    this.is_show ? 'show' : ''
+                    {
+                        fade: !this.noFade,
+                        show: this.is_show,
+                    }
                 ];
             },
             headerClasses() {
                 return [
                     'modal-header',
                     // Rounding is needed to fix a bug in bootstrap V4.beta.1 CSS
-                    Boolean(this.headerBgVariant) ? 'rounded-top' : '',
-                    Boolean(this.headerBgVariant) ? `bg-${this.headerBgVariant}` : '',
-                    Boolean(this.headerTextVariant) ? `text-${this.headerTextVariant}` : '',
-                    Boolean(this.headerBorderVariant) ? `border-${this.headerBorderVariant}` : ''
+                    {
+                        'rounded-top': this.headerBgVariant,
+                        [`bg-${this.headerBgVariant}`]: this.headerBgVariant,
+                        [`text-${this.headerTextVariant}`]: this.headerTextVariant,
+                        [`border-${this.headerBorderVariant}`]: this.headerBorderVariant,
```

Comment:
Should these be wrapping the variants in `Boolean(...)` (since they are strings)

Reply:
i.e
```js
[`bg-${this.headerBgVariant}`]: Boolean(this.headerBgVariant),
```

Reply:
And similar for the other classes sections as well...

Reply:
Like we discussed, Vue checks these in an `if` condition, so it isn't necessary. You're thought and them getting cached in the computed props is smart though, so best to cast them.